java/compaction-job-execution/src/main/java/sleeper/compaction/jobexecution/CompactSortedFilesRunner.java

```diff
@@ -72,6 +80,7 @@ public class CompactSortedFilesRunner {
     private final TablePropertiesProvider tablePropertiesProvider;
     private final StateStoreProvider stateStoreProvider;
     private final CompactionJobStatusStore jobStatusStore;
+    private final List<CompactionJobStatus> finishedJobs;
```

Comment:
I'm concerned that if a task runs for a very long time (which is possible) then this list will keep growing in size, and eventually cause an out-of-memory error. It would be better to store the running totals from the jobs, rather than storing all the information about each job.